### ISSUE TITLE
ReferenceError: match is not defined

### ISSUE THREAD

**Issue by chrishiste** [krisk/Fuse#162]
Hello, I'm trying to use the key options but it throws "ReferenceError: match is not defined". Here is the code I'm using to test. I'm using fuse 3.0.3. It is fairly basic so I suppose there is something wrong in fuse code am I right ?  

```js
 const data = [
    {
      name: 'A1',
      type: 'A2',
    },
    {
      name: 'B1',
      type: 'B2',
    },
    {
      name: 'C1',
      type: 'C2',
    },
  ];
  const options = {
    keys: ['name', 'type'],
  };

  const fuse = new Fuse(data, options);
  const result = fuse.search(textToSearch);
```

**Comment by jGeek314** [comment]
Just curious why you closed this.  I think there's still a bug in bitap_regex_search.js

**Comment by chrishiste** [comment]
@jGeek314 Ah sorry, there is indeed still a bug in fuse.js code. Unfortunately I don't really have time to dig in the codebase. The problem only occured with really long strings (containing Regex special characters like parenthesis or brackets). I suppose there is a if statement and it's not declaring?/calling? match correctly. I closed the issue because I tried previous version of fuse.js there was a similar problem but not the same error message. From there I thought my code must be wrong and the only wrong thing could only be the input I'm sending to fuse. So I'm now modifying my input first.
